### PR TITLE
Add safeties, WholeBodyControl config for LCM2ROSControl, and better hand joint limits

### PR DESCRIPTION
**Pull request by gizatt**
Hands behave poorly if fingers are driven to positions lower than ~0.5, so I've added that as the lower bound for the moment.

Adds force and position bounding for LCM2ROSControl to prevent it from applying forces out of bounds. We need to populate more precise force and joint limits per-joint; right now defaults to the old |q| < 3.14 and |force| < 1000.

And adds some general options enabling us to use LCM2ROSControl for running the MIT controller.

### CODE REVIEW

File: config/HandControl.yaml [hunk 1]
```diff
@@ -7,32 +7,32 @@ HandPositionGoalController:
...
         leftIndexFingerMotorPitch1:
             has_position_limits: true
             min_position: 0.5
-            max_position: 2.0
+            max_position: 2.5
         leftMiddleFingerMotorPitch1:
             has_position_limits: true
             min_position: 0.5
-            max_position: 2.0
+            max_position: 2.7
```
Comment (wxmerkt) [inline config/HandControl.yaml:31]: That's likely to be dangerous for Edinburgh - cc @VladimirIvan (need separate hand-dependant config)

File: config/HandControl.yaml [hunk 2]
```diff
@@ -7,32 +7,32 @@ HandPositionGoalController:
...
         leftPinkyMotorPitch1:
             has_position_limits: true
             min_position: 0.5
-            max_position: 2.0
+            max_position: 2.7
```
Comment (wxmerkt) [inline config/HandControl.yaml:35]: similarly

**Comment by wxmerkt** [comment]
Thanks, that's amazing. I believe it would be time to have separate `HandControl.yaml` configuration files as especially the `max_position` depends on the tendons installed in the hands and we'd calibrated them to what's tendon-safe (that said, we currently have separate files for the left and right hands, so far without PR/merge). I wonder, what's the behaviour that you've seen when the `min_position` was `0.0` instead of `0.5`/what are the differences?

**Comment by manuelli** [comment]
I resolved the merge conflict and am going to go ahead and merge this. We can keep iterating on these classes but I want to get everything into master for now.

+1

File: config/HandControl.yaml [hunk 3]
```diff
@@ -7,65 +7,65 @@ HandPositionGoalController:
...
-            min_position: 0.0
-            max_position: 2.0
+            min_position: 0.5
+            max_position: 2.5
         leftMiddleFingerMotorPitch1:
             has_position_limits: true
-            min_position: 0.0
-            max_position: 2.0
+            min_position: 0.5
+            max_position: 2.7
```
Comment (wxmerkt) [inline config/HandControl.yaml:31]: That's likely to be dangerous for Edinburgh - cc @VladimirIvan (need separate hand-dependant config)

(NB: again as GitHub shows the previous comment as already resolved)

**Comment by wxmerkt** [comment]
@manuelli would you mind elaborating on what difference in behaviour you saw between 0.5/0.0 as the minimum joint limit; curious to see whether we've been seeing the same

**Comment by gizatt** [comment]
I'm the one who's been doing the hand work -- might have accidentally
posted as lucas at some point. I was advised by NASA that going near 0.0 /
below 0.0 creates a chance that the tendon will start winding in reverse,
which causes the uC to become confused why the finger is closing, causing
the motor to wind some more, which quickly cuts the tendon on an internal
wall. I haven't seen a tendon break for this reason yet but I've been being
very cautious about it.

Understood about danger of joint limits, Lucas is moving quickly because we
want to get our controller up tonight. I'll separate out hand control files
and restore your old values to one of them. It'll be worth sanity-checking
them on your own hardware beforehand, though.

**Comment by wxmerkt** [comment]
Thanks for the explanation Greg :)! Great insights (we didn't get that advice), we've seen some strange behaviour when commanding 0.0s (i.e. a finger suddenly and rapidly closing all the way until the tendon snaps - happened twice so far). We'll be avoiding the lower limit going forward as well - thanks for the insight.

Good luck with getting the controller up tonight! :)

**Pull request merged**